pkg/resourceconverter/resource_converter.go

```diff
@@ -22,53 +23,87 @@ type TagField struct {
 	Value string `yaml:"value"`
 }
 
+func (f TagField) IsZero() bool {
+	return f.Name == ""
+}
+
 type ReflectionConverter struct {
-	ResourceType string
-	TagField     TagField
-	IdField      string
-	Region       string
+	ResourceType   string
+	TagField       TagField
+	IdField        string
+	DisplayIdField string
+	Region         string
 }
 
 func (rc *ReflectionConverter) ToResource(ctx context.Context, x any, tags model.Tags) (model.Resource, error) {
-	t := reflect.TypeOf(x)
-
-	// get the id field
-	var id string
-	for i := 0; i < t.NumField(); i++ {
-		field := t.Field(i)
-		name := field.Name
-		if name == rc.IdField {
-			fieldPtrRef := reflect.ValueOf(x).FieldByName(name)
-			fieldRef := reflect.Indirect(fieldPtrRef)
-			if !fieldRef.IsZero() {
-				id = fmt.Sprintf("%v", fieldRef.Interface())
-			}
-			break
-		}
+	resource := model.Resource{
+		Region: rc.Region,
+		Type:   rc.ResourceType,
 	}
 
-	if id == "" {
+	resource.Id = rc.findId(rc.IdField, x)
+	if resource.Id == "" {
 		return model.Resource{}, fmt.Errorf("could not find id field '%v' for type '%v", rc.IdField, rc.ResourceType)
 	}
 
+	if err := rc.loadDisplayId(x, &resource); err != nil {
+		return model.Resource{}, err
+	}
+
 	// generate tags field
-	if tags == nil {
+	if !rc.TagField.IsZero() {
 		//use field
 		tagsValue := reflect.ValueOf(x).FieldByName(rc.TagField.Name)
 		if !tagsValue.IsValid() {
-			return model.Resource{}, fmt.Errorf("Could not find tag field '%v' for type '%v", rc.TagField.Name, rc.ResourceType)
+			return model.Resource{}, fmt.Errorf("could not find tag field '%v' for type '%v", rc.TagField.Name, rc.ResourceType)
 		}
-		tags = getTags(tagsValue, rc.TagField)
+		resource.Tags = getTags(tagsValue, rc.TagField)
 	}
+
+	resource.Tags = append(resource.Tags, tags...)
+
 	marshaledStruct, err := json.Marshal(x)
 	if err != nil {
 		return model.Resource{}, err
 	}
-	return model.Resource{
-		Id:      id,
-		Region:  rc.Region,
-		Type:    rc.ResourceType,
-		RawData: marshaledStruct,
-		Tags:    tags,
-	}, nil
+
+	resource.RawData = marshaledStruct
+
+	return resource, nil
+}
+
+func (rc *ReflectionConverter) loadDisplayId(x any, resource *model.Resource) error {
+	if rc.DisplayIdField == "" {
+		return nil
+	}
+
+	id := rc.findId(rc.DisplayIdField, x)
+
+	if id == "" {
+		return fmt.Errorf("could not find display id field '%v' for type '%v", rc.DisplayIdField, rc.ResourceType)
+	}
+
+	resource.DisplayId = id
+
+	return nil
+}
+
+func (rc *ReflectionConverter) findId(fieldName string, x any) string {
+	t := reflect.TypeOf(x)
+
+	var id string
+	for i := 0; i < t.NumField(); i++ {
+		field := t.Field(i)
+		name := field.Name
+		if name == fieldName {
+			fieldPtrRef := reflect.ValueOf(x).FieldByName(name)
+			fieldRef := reflect.Indirect(fieldPtrRef)
+			if !fieldRef.IsZero() {
+				id = fmt.Sprintf("%v", fieldRef.Interface())
+			}
+			break
+		}
+	}
```

Comment:
you can also do directory and use a recover https://gobyexample.com/recover if it panics - it will avoid the extra for loop. 
`fieldPtrRef := reflect.ValueOf(x).FieldByName(name)`

Reply:
I was thinking for resources with a lot of attributes this could be a bit this for loop could become a bottleneck - but please disregard if this is not an issue.

if we wanted to test this with the load tests, we would need to add more properties in the `rawData` of the template used by the load-test `pkg/datastore/testdata/resources.json`
probably not a huge one performance wise - but since `reflect` is usually considered slow and we would call `t.Field(i)` for each property it got me thinking about it.

Reply:
actually nevermind the load test would not use resource converter.
so we would need a dedicated load test for this.

Reply:
1. Using panic/recover for control flow is generally bad form in Go, so that's not really something I would want to do.
2. `reflect.Value.FieldByName` only panics if the value isn't a struct, not if the field isn't found, so we wouldn't be able to rely on the panic anyways.
3. The returned value is a zero value if the field isn't found, which is easy to test for.

I took a closer look at this function (which was just moved from elsewhere in the file), and yes, it can be done more cleanly and without the loop. I've cleaned up the implementation.